backend/app/controllers/products_controller.rb

```diff
@@ -0,0 +1,65 @@
+class ProductsController < ApplicationController
+    include Dry::Monads::Result::Mixin
+
+    #POST /products
+    def create
+        case product_creation
+        in Success(product)
+          render json: prepare_json(product), status: :created
+        in Failure(error)
+          render json: { errors: error.messages }, status: :unprocessable_entity
+        end
+    end
+
+    #PUT/PATCH /products/edit/1
+    def update
```

Comment:
Will this method really update the product? Looks like 'set_product` do nothing except finding product.